### PR TITLE
feat(connector): [iatapay] fix refund amount, hardcode IN for UPI, send merchant payment id

### PR DESCRIPTION
**Pull request by arvindpatel24**
## Type of Change

- [x] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
fix refund amount, hardcode IN for UPI, send merchant payment id

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [x] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Review by ArjunKarthik — DISMISSED** [review]

**Review by jagan-jaya — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

File: crates/router/src/connector/iatapay/transformers.rs [hunk 1]
```diff
@@ -81,7 +83,11 @@ pub struct IatapayPaymentsRequest {
 impl TryFrom<&types::PaymentsAuthorizeRouterData> for IatapayPaymentsRequest {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
-        let country = item.get_billing_country()?.to_string();
+        let payment_method = item.payment_method;
+        let country = match payment_method {
+            PaymentMethod::Upi => "IN".to_string(),
```
Comment (Abhicodes-crypto) [inline crates/router/src/connector/iatapay/transformers.rs:88]: Hardcoding is not the right way. Can you create an issue to tackle this later.

**Review by Abhicodes-crypto — APPROVED** [review]
Minor comments, other than that LGTM.

File: crates/router/src/connector/iatapay/transformers.rs (hunk 1 above)
Comment (arvindpatel24) [reply crates/router/src/connector/iatapay/transformers.rs:88]: Hey, this is merchant requirement to hardcode country as UPI is only applicable in India. cc: @shakthi-22

File: crates/router/src/connector/iatapay/transformers.rs [hunk 2]
```diff
@@ -246,8 +253,8 @@ impl<F, T>
 #[serde(rename_all = "camelCase")]
 pub struct IatapayRefundRequest {
     pub merchant_id: Secret<String>,
-    pub merchant_refund_id: String,
-    pub amount: i64,
+    pub merchant_refund_id: Option<String>,
```
[the remark this answers is not in the document]
Comment (arvindpatel24) [reply crates/router/src/connector/iatapay/transformers.rs:256]: It is non-mandatory field in connector refund request, @anji-reddy-j missed this.

**Pull request merged**